src/core/shared/types.d.ts

```diff
@@ -83,6 +90,8 @@ export type PostHookDocumentationBundle = {
 
 // Configurable Hooks
 
+// TODO: For these hooks, at this stage the path where files are written is not changeable, so let's create new types that account for that.
+
```

Comment:
Also let's now export a function that allows for the DocPageReferences to be changeable, which is how where the files end up can be configured

Reply:
This works now, now I need to make sure that no matter what URL I throw at it when using TransformReference, things work fine.

I think I should define some method that can find the "root" and then all URLs should be root based